packages/xrpl/src/sugar/submit.ts

```diff
@@ -176,6 +247,94 @@ async function waitForFinalTransactionOutcome(
   )
 }
 
+async function submitAndWaitHelper(
+  client: Client,
+  transaction: Transaction | string,
+  opts?: {
+    // If true, autofill a transaction.
+    autofill?: boolean
+    // If true, and the transaction fails locally, do not retry or relay the transaction to other servers.
+    failHard?: boolean
+    // A wallet to sign a transaction. It must be provided when submitting an unsigned transaction.
+    wallet?: Wallet
+  },
+): Promise<TxResponse> {
+  const signedTx = await getSignedTx(client, transaction, opts)
+
+  const lastLedger = getLastLedgerSequence(signedTx)
+  if (lastLedger == null) {
+    throw new ValidationError(
+      'Transaction must contain a LastLedgerSequence value for reliable submission.',
+    )
+  }
+
+  const response = await submitRequest(client, signedTx, opts?.failHard)
+
+  const txHash = hashes.hashSignedTx(signedTx)
+  return waitForFinalTransactionOutcome(
+    client,
+    txHash,
+    lastLedger,
+    response.result.engine_result,
+  )
+}
+
+// eslint-disable-next-line max-params, max-lines-per-function -- necessary
+async function submitAndWaitBatchHelper(
+  client: Client,
+  account: string,
+  batchMap: Map<
```

Comment:
It would be a little neater if `submitAndWaitBatchHelper` only took `batchMap.get(account)` as a param, instead of the whole map